File: queues/ibmmq/Python/tests/test_ibmmqclient.py
import sys
from pathlib import Path
from unittest.mock import MagicMock, patch

import pytest

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
import ibmmqclient


@pytest.fixture(autouse=True)
def env(monkeypatch):
    monkeypatch.setenv("MQ_HOST", "mq.local")
    monkeypatch.setenv("MQ_QUEUE_MANAGER", "QM1")
    monkeypatch.setenv("MQ_QUEUE", "DEV.QUEUE")


def test_init_missing_host(monkeypatch):
    monkeypatch.delenv("MQ_HOST", raising=False)
    assert ibmmqclient.Init()["status_code"] == 601


def test_put():
    mock_pymqi = MagicMock()
    qmgr = MagicMock()
    queue = MagicMock()
    mock_pymqi.connect.return_value = qmgr
    mock_pymqi.Queue.return_value = queue
    mock_pymqi.CMQC = MagicMock()
    mock_pymqi.CMQC.MQCHT_CLNT = 1
    mock_pymqi.CMQC.MQXPT_TCP = 2
    mock_pymqi.CD.return_value = MagicMock()
    with patch.dict(sys.modules, {"pymqi": mock_pymqi}):
        r = ibmmqclient.Put({"id": 1})
    assert r == {"status_code": 200}

File: queues/ibmmq/C/tests/test_ibmmqclient.c
#include "../../../../tests/c/testutil.h"
#include "../include/ibmmqclient.h"
#include <stdlib.h>

TEST(test_init_missing)
{
    unsetenv_safe("MQ_HOST");
    char *json = ibmmq_init();
    ASSERT("601", json_status(json) == 601);
    free(json);
}

TEST(test_init_ok)
{
    setenv("MQ_HOST", "mq.local", 1);
    setenv("MQ_QUEUE_MANAGER", "QM1", 1);
    setenv("MQ_QUEUE", "Q1", 1);
    char *json = ibmmq_init();
    ASSERT("200", json_status(json) == 200);
    free(json);
}

int main(void)
{
    RUN(test_init_missing);
    RUN(test_init_ok);
    printf("ibmmq C tests passed\n");
    return 0;
}
